src/main/java/com/metamx/http/client/response/ToStringResponseHandler.java

```diff
@@ -35,21 +36,22 @@ public ToStringResponseHandler(Charset charset)
   @Override
   public ClientResponse<StringBuilder> handleResponse(HttpResponse response)
   {
-    return ClientResponse.unfinished(new StringBuilder(response.getContent().toString(charset)));
+    ByteBuf content = response instanceof HttpContent ? ((HttpContent) response).content() : Unpooled.EMPTY_BUFFER;
```

Comment:
Suggested if-else and provide `""` empty string directly instead of converting empty buffer to empty string